internal/volumehelper/volume_policy_helper.go

```diff
@@ -0,0 +1,245 @@
+			}
+			// now fetch the PV and call GetMatchAction on it
+			pv, err := v.GetPVForPVC(pvc)
+			if err != nil {
+				return FSBackupActionMatchingVols, FSBackupNonActionMatchingVols, NoActionMatchingVols, err
+			}
+			// now get the action for pv
+			action, err := volumePolicies.GetMatchAction(pv)
+			if err != nil {
+				return FSBackupActionMatchingVols, FSBackupNonActionMatchingVols, NoActionMatchingVols, err
+			}
+
+			// Record volume list having at least some action so that they are not processed in legacy fallback option
+			if action == nil {
+				NoActionMatchingVols = append(NoActionMatchingVols, vol.Name)
+			}
+
+			// Now if the matched action is not nil and is `fs-backup` then add that Volume to the FSBackupActionMatchingVols
+			// else add that volume to FSBackupNonActionMatchingVols
+			// we already tracked the volume not matching any kind actions supported by volume policy in  NoActionMatchingVols
+			// The NoActionMatchingVols list will be processed via legacy annotation based approach as a fallback option
+			if action != nil && action.Type == resourcepolicies.FSBackup {
+				FSBackupActionMatchingVols = append(FSBackupActionMatchingVols, vol.Name)
+			} else {
+				FSBackupNonActionMatchingVols = append(FSBackupNonActionMatchingVols, vol.Name)
+			}
+		}
+	}
+	return FSBackupActionMatchingVols, FSBackupNonActionMatchingVols, NoActionMatchingVols, nil
+}
+
+func (v *VolumeHelperImpl) ShouldIncludeVolumeInBackup(vol corev1api.Volume, volsToExclude []string) bool {
+	includeVolumeInBackup := true
+	// cannot backup hostpath volumes as they are not mounted into /var/lib/kubelet/pods
+	// and therefore not accessible to the node agent daemon set.
+	if vol.HostPath != nil {
+		includeVolumeInBackup = false
+	}
+	// don't backup volumes mounting secrets. Secrets will be backed up separately.
+	if vol.Secret != nil {
+		includeVolumeInBackup = false
+	}
+	// don't backup volumes mounting ConfigMaps. ConfigMaps will be backed up separately.
+	if vol.ConfigMap != nil {
+		includeVolumeInBackup = false
+	}
+	// don't backup volumes mounted as projected volumes, all data in those come from kube state.
+	if vol.Projected != nil {
+		includeVolumeInBackup = false
+	}
+	// don't backup DownwardAPI volumes, all data in those come from kube state.
+	if vol.DownwardAPI != nil {
+		includeVolumeInBackup = false
+	}
+	if vol.PersistentVolumeClaim != nil && v.BackupExcludePVC {
+		includeVolumeInBackup = false
+	}
+	// don't backup volumes that are included in the exclude list.
+	if util.Contains(volsToExclude, vol.Name) {
+		includeVolumeInBackup = false
+	}
+	// don't include volumes that mount the default service account token.
+	if strings.HasPrefix(vol.Name, "default-token") {
+		includeVolumeInBackup = false
+	}
+	return includeVolumeInBackup
+}
+
+func (v *VolumeHelperImpl) GetPVCForPodVolume(vol corev1api.Volume, pod *corev1api.Pod) (*corev1api.PersistentVolumeClaim, error) {
```

Comment:
This looks like a common function, can we add this into util package?

Reply:
Done